src/components/AlreadyConnectedModal.tsx

```diff
@@ -0,0 +1,64 @@
+import { Dialog } from "@headlessui/react";
+import { useState } from "react";
+import { PublicUser, User } from "../utils/types";
+
+interface AlreadyConnectedModalProps {
+  // represents the 'me', the user trying to connect to someone
+  currentUser: User;
+  // represents the other user 'I' am trying to connect to.
+  userToConnectTo: PublicUser;
+
+  closeModal: () => void;
+  handleManageRequest: () => void;
+}
+
+const AlreadyConnectedModal = (
+  props: AlreadyConnectedModalProps
+): JSX.Element => {
+  const [isOpen, setIsOpen] = useState(true);
+
+  const onClose = () => {
+    setIsOpen(false);
+    props.closeModal();
+  };
+  return (
+    <Dialog open={isOpen} onClose={onClose} className="relative z-50">
+      {/* backdrop panel */}
+      <div className="fixed inset-0 backdrop-blur-sm" aria-hidden="true">
+        {/* Full-screen container to center the panel */}
+        <div className="fixed inset-0 flex items-center justify-center p-4">
+          {/* dialog panel container  */}
+          <Dialog.Panel className="justify-center rounded-md shadow-md bg-white h-3/6 w-3/6 content-center flex flex-col p-9 gap-4">
+            <Dialog.Title className="font-bold text-2xl text-center">
+              You already have a pending request from this user
+            </Dialog.Title>
+            <div className="text-sm">
+              {props.userToConnectTo.preferredName} has already sent you a
+              connection request. Click the manage request button below to
+              accept or decline the request!
```

Comment:
Very minor but do we want this exclamation mark here, idk it just sounds very extra imo